llvm/test/lab2/korablev_nikita/shift.ll

```diff
@@ -0,0 +1,121 @@
+; RUN: opt -load-pass-plugin %llvmshlibdir/Replace-Mult-Shift_Korablev-Nikita_FIIT3%pluginext\
+; RUN: -passes=korablev-replace-mul-shift -S %s | FileCheck %s
+
+; int test1() {
+;     int b = 3;
+;     int res = b * 4;
+;     return res;
+; }
+
+; int test2() {
+;     int b = 3;
+;     int res = 4 * b;
+;     return res;
+; }
+
+; int test3(int a) {
+;     int res = 4 * a;
+;     return res;
+; }
+
+; int test4() {
+;     int a = 3;
+;     int res = 4 + a;
+;     return res;
+; }
+
+; int test5() {
+;     int a = 3;
+;     return a * 4;
+; }
+
+; int test6() {
+;     int a = 3;
+;     return 4 * a;
+; }
+
+define dso_local noundef i32 @_Z5test1v() #0 {
+entry:
+  %b = alloca i32, align 4
+  %res = alloca i32, align 4
+  store i32 3, ptr %b, align 4
+  %0 = load i32, ptr %b, align 4
+  %mul = mul nsw i32 %0, 4
+  store i32 %mul, ptr %res, align 4
+  %1 = load i32, ptr %res, align 4
+  ret i32 %1
+}
+
+; CHECK-LABEL: @_Z5test1v
+; CHECK: %shiftInst = shl i32 4, %0
```

Comment:
`shl i32 4, %0` is equal to (4 << %0) = (4 << 3) = 32
Whilst `mul nsw i32 %0, 4` is equal to (%0 * 4) = (3 * 4) = 12